microcosm_postgres/temporary/methods.py

```diff
@@ -0,0 +1,42 @@
+"""
+Extension methods on top of a temporary table.
+
+"""
+from microcosm_postgres.context import SessionContext
+from sqlalchemy.dialects.postgresql import insert
+
+
+def insert_many(self, items):
+    """
+    Insert many items at once into a temporary table.
+
+    Items are expected to extend `IdentityMixin`
+
+    """
+    return SessionContext.session.execute(
+        self.insert(values=[
+            item._members()
+            for item in items
+        ]),
+    ).rowcount
+
+
+def upsert_into(self, table):
+    """
+    Upsert from a temporarty table into another table.
+
+    """
+    return SessionContext.session.execute(
+        insert(table).from_select(
+            self.c,
+            self,
+        ).on_conflict_do_nothing(),
```

Comment:
worth customizing the conflict clause?

Reply:
I don't think this makes sense right now. Constructing the configuration for `on_conflict_do_update` is non-trivial and I don't know what sort of customization API we'd want here without a few use cases. And I don't have any use cases yet for conflict updates with temporary tables.